app/src/main/java/com/digitalarchitects/rmc_app/data/editmyaccount/EditMyAccountViewModel.kt

```diff
@@ -1,118 +1,228 @@
 package com.digitalarchitects.rmc_app.data.editmyaccount
 
-import androidx.compose.runtime.Composable
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import androidx.navigation.NavController
-import androidx.navigation.compose.rememberNavController
 import com.digitalarchitects.rmc_app.app.RmcScreen
+import com.digitalarchitects.rmc_app.model.UserType
 import com.digitalarchitects.rmc_app.room.UserDao
 import com.digitalarchitects.rmc_app.room.UserTable
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.SharingStarted
-import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
 
 
 class EditMyAccountViewModel(
-    private val dao: UserDao,
-    private val navController: NavController
-    ): ViewModel() {
+    private val userDao: UserDao,
+) : ViewModel() {
+    private val _navigateToScreen = MutableStateFlow<RmcScreen?>(null)
+    val navigateToScreen = _navigateToScreen.asStateFlow()
 
     private val _state = MutableStateFlow(EditMyAccountUIState())
-    val state = _state.stateIn(
-        viewModelScope,
-        SharingStarted.WhileSubscribed(5000),
-        EditMyAccountUIState()
-    )
-    fun onEvent(event: EditMyAccountUIEvent){
-        when(event){
+    private val _uiState = _state
+    val uiState: StateFlow<EditMyAccountUIState> get() = _uiState.asStateFlow()
+
+    fun onEvent(event: EditMyAccountUIEvent) {
+        when (event) {
+            is EditMyAccountUIEvent.ShowUser -> {
+                try {
+                    runBlocking {
```

Comment:
Waarom heb je ervoor gekozen om runBlocking te gebruiken?

Reply:
Zag het gebruikt worden in een voorbeeld en philip lackner filmpje. Echter is dit al achterhaald en zou je dit eigenlijk niet in deze situatie willen gebruiken. zie ook de beschrijving -> Todo.